### ISSUE TITLE
Synergy Engine: Implement card embeddings for semantic similarity

### ISSUE THREAD

**Issue by RdHamilton** [RdHamilton/MTGA-Companion#828]
## Overview
Train card embeddings where cards that appear in similar decks/contexts are close in vector space. This enables finding cards similar to a given card that also synergize with the deck's strategy.

## Concept
Similar to word2vec, but for Magic cards:
- Cards in same decks → similar embeddings
- Cards with similar roles → similar embeddings
- Can find "cards like X that synergize with Y"

## Training Data
1. **Deck co-occurrence**: Cards appearing together in decks
2. **Replacement patterns**: When players swap cards, those cards are similar
3. **Oracle text**: Cards with similar abilities should be close

### Option A: Train Custom Embeddings
```python
# Using deck data as "sentences" and cards as "words"
from gensim.models import Word2Vec

# Each deck is a "sentence" of card IDs
decks = [
    [card_id_1, card_id_2, ...],  # Deck 1
    [card_id_3, card_id_4, ...],  # Deck 2
]

model = Word2Vec(decks, vector_size=64, window=5, min_count=5)
```

### Option B: Use Pre-trained Text Embeddings
1. Generate text description for each card (oracle + type + stats)
2. Use sentence-transformers to embed descriptions
3. Similar cards = similar descriptions

### Option C: Hybrid Approach
- Combine deck co-occurrence embeddings with oracle text embeddings
- Weight by usefulness for synergy prediction

## Go Integration
```go
type CardEmbedding struct {
    CardID    int
    Vector    []float64  // 64-128 dimensions
    UpdatedAt time.Time
}

func (e *EmbeddingService) GetSimilarCards(cardID int, topK int) []SimilarCard {
    // Find nearest neighbors in embedding space
}

func (e *EmbeddingService) GetSynergyScore(cardA, cardB int) float64 {
    // Cosine similarity between embeddings
}
```

## Storage Options
1. **SQLite with JSON**: Store vectors as JSON arrays
2. **Separate vector DB**: Use Qdrant, Milvus, or similar
3. **In-memory**: Load embeddings on startup

## Training Pipeline
1. Fetch deck data from Archidekt/Moxfield
2. Preprocess: Remove basic lands, normalize card names
3. Train embeddings (Python script)
4. Export to JSON/SQLite
5. Load in Go application

## Use Cases
1. **"Cards like X"**: Find similar cards for budget replacements
2. **Synergy scoring**: Cosine similarity as synergy factor
3. **Archetype clustering**: Group cards by play pattern
4. **Anomaly detection**: Find novel synergies (cards that don't usually go together but might work)

## Acceptance Criteria
- [ ] Choose embedding approach (custom vs pre-trained)
- [ ] Build training pipeline
- [ ] Train embeddings on 10,000+ decks
- [ ] Implement similarity search in Go
- [ ] Integrate into recommendation engine
- [ ] Evaluate embedding quality (are similar cards actually similar?)
